File: examples/python/advanced/serialization/custom-objects.py
from aws_durable_execution_sdk_python import DurableContext, durable_execution

class Order:
    def __init__(self, order_id: str, amount: float, customer: str):
        self.order_id = order_id
        self.amount = amount
        self.customer = customer
    
    def to_dict(self) -> dict:
        return {
            "order_id": self.order_id,
            "amount": self.amount,
            "customer": self.customer
        }
    
    @classmethod
    def from_dict(cls, data: dict) -> "Order":
        return cls(
            order_id=data["order_id"],
            amount=data["amount"],
            customer=data["customer"]
        )

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    order = Order("ORD-123", 99.99, "Jane Doe")
    
    # Convert to dict before passing to step
    result = context.step(process_order, order.to_dict())
    return result

File: examples/java/advanced/serialization/custom-objects.java
// POJO with Jackson annotations (serializes automatically)
public class Order {
    private String orderId;
    private double amount;
    private String customer;

    public Order() {}

    public Order(String orderId, double amount, String customer) {
        this.orderId = orderId;
        this.amount = amount;
        this.customer = customer;
    }

    // Getters and setters
    public String getOrderId() { return orderId; }
    public double getAmount() { return amount; }
    public String getCustomer() { return customer; }
}

// Usage in handler
var order = new Order("ORD-123", 99.99, "Jane Doe");
var result = ctx.step("process-order", Map.class, stepCtx -> processOrder(order));
